File: LeetCode/Java/Easy/1021_remove_outermost_parentheses.java
class Solution {
    public String removeOuterParentheses(String S) {
        StringBuilder ans = new StringBuilder();
        int open = 0, close = 0, start = 0;
        for(int i = 0; i < S.length(); i++){
            if(S.charAt(i) == '(')
                open++;
            else
                close++;
            if(open == close){
                ans.append(S.substring(start + 1, i));
                start = i + 1;
            }
        }
        return ans.toString();
    }
}

File: LeetCode/Python/Easy/1021_remove_outermost_parentheses.py
class Solution:
    def removeOuterParentheses(self, S: str) -> str:
        def checkVal(ans, val, paren):
            if val != 0:
                ans.append(paren)
            return ans
        ans = []
        val = 0
        for paren in S:
            if paren == '(':
                checkVal(ans, val, paren)
                val += 1
            else:
                val -= 1
                checkVal(ans, val, paren)
        return "".join(ans)
